### ISSUE TITLE
DoubleInRange can return invalid result

### ISSUE THREAD

**Issue by agroce** [trailofbits/deepstate#249]
Exposing harness:

```
TEST(DoubleGen, Ranges) {
  double low = DeepState_Double();
  double high = DeepState_Double();
  if (low > high) {
    double temp = low;
    low = high;
    high = temp;
  }
  double v = DeepState_DoubleInRange(low, high);
  printf ("BETWEEN %lf and %lf = %lf\n", low, high, v);
  ASSERT (v >= low) << v << " < " << low;
  ASSERT (v <= high) << v << " > " << high;
}
```

```
TRACE: Initialized test input buffer with data from `badrange/626a94bf5417a8612276dcf16f1b3732c2dc661c.fail`
TRACE: Running: DoubleGen_Ranges from flcheck.cpp(21)
TRACE: BETWEEN -5391589316425305909279862304866087899446200827463439442055990902868051109768836733245644799125247197011587549134886469858364808782646287823479839598713335752992558075594606684340124321526882109512440548378783057084214152508118977971073597833216.000000 and 783633934227438932531297092923645530996736.000000 = 8303292228502777799233726806010584231907422109696.000000

CRITICAL: flcheck.cpp(32): 8303292228502777799233726806010584231907422109696.000000 > 783633934227438932531297092923645530996736.000000
ERROR: Failed: DoubleGen_Ranges
```

**Comment by agroce** [comment]
@pgoodman take a look at the code and see if you can determine the issue, when you get a chance.  I just do a bool split to either go from low to -0 or 0 to high, when the range crosses a sign boundary

**Comment by agroce** [comment]
Fixed in #250
